numpy/core/tests/test_scalar_methods.py

```diff
@@ -0,0 +1,79 @@
+"""
+Test the scalar constructors, which also do type-coercion
+"""
+from __future__ import division, absolute_import, print_function
+
+import fractions
+import numpy as np
+
+from numpy.testing import (
+    run_module_suite,
+    assert_equal, assert_almost_equal, assert_raises, assert_warns,
+    dec
+)
+
+float_types = [np.half, np.single, np.double, np.longdouble]
+
+def test_float_as_integer_ratio():
+    # derived from the cpython test "test_floatasratio"
+    for ftype in float_types:
+        for f, ratio in [
+                (0.875, (7, 8)),
+                (-0.875, (-7, 8)),
+                (0.0, (0, 1)),
+                (11.5, (23, 2)),
+            ]:
+            assert_equal(ftype(f).as_integer_ratio(), ratio)
+
+        rstate = np.random.RandomState(0)
+        fi = np.finfo(ftype)
+        for i in range(1000):
+            exp = rstate.randint(fi.minexp, fi.maxexp - 1)
+            frac = rstate.rand()
+            f = np.ldexp(frac, exp, dtype=ftype)
+
+            n, d = f.as_integer_ratio()
+
+            try:
+                dn = np.longdouble(str(n))
+                df = np.longdouble(str(d))
+            except (OverflowError, RuntimeWarning):
+                # the values may not fit in any float type
+                continue
+
+            assert_equal(
+                dn / df, f,
+                "{}/{} (dtype={})".format(n, d, ftype.__name__))
+
+        R = fractions.Fraction
```

Comment:
The nesting makes the tests below a little hard to follow -- only need `ftype` iteration / parametrization here I think?